tests/mesolve/me_solver_tester.py

```diff
@@ -82,80 +75,47 @@ def _test_adjoint(
         system: OpenSystem,
         *,
         num_t_save: int,
-        rtol: float = 1e-5,
-        atol: float = 3e-3,
+        rtol_rho: float = 1e-5,
+        rtol_exp: float = 1e-5,
+        atol_rho: float = 3e-3,
+        atol_exp: float = 1e-2,
```

Comment:
I think we're defaulting these in the wrong way. What we want is to test whether the gradients are close but in relative value rather than in absolute value. So, I think we should raise the `rtol` quite a lot, and maybe lower the `atol`. This would also help standardize the values between the `exp` check and `rho` check and between Rouchon1 and Rouchon2.

Typically, I think 5% relative error on the gradients is fine, so `rtol=5e-2`, while we should maybe only tolerate ~0.001-0.003 absolute error (for very small gradients close to 0, e.g. when the gradient descent is converging to a global minima), so `atol = 3e-3`. This way all tests pass with the same tolerences. 

Then, you can lower these for Rouchon2 if you want (e.g. `rtol=1e-2` and `atol=1e-3`), but I don't think that's required.